pennylane/templates/layers/basic_entangler.py

```diff
@@ -23,12 +23,14 @@
     check_no_variable,
     check_number_of_layers,
     get_shape,
+    check_type,
+    check_shapes,
 )
 from pennylane.wires import Wires
 
 
 @template
-def BasicEntanglerLayers(weights, wires, rotation=None):
+def BasicEntanglerLayers(weights, wires, rotation=None, interactions=None):
```

Comment:
I think it might make sense to leverage `broadcast` more directly, so that these changes proposed changes can be used outside of just `BasicEntanglerLayers`. 

What I'd suggest is the following:
- change the keyword arg here to "pattern" (same as `qml.broadcast`)
- add this new functionality (ability to arrange two-qubit gates according to a custom user-specified pattern) into `qml.broadcast`
- in this template, rely on that implementation

Reply:
Yeah, I was talking to Maria in the thread of the original issue, and she says that it makes more sense to just make an entirely new template, `EntanglerLayers`, so that the "basic-ness" of this template can be preserved.

Reply:
Yes @co9olguy 's suggestion would make a lot of sense. 

Upon second thought, if the ``pattern`` argument has a default, a user would never have to touch it if the template is just used in its basic form...and since the entanglers do not take parameters it is still pretty simple to use.

@Lucaman99 if you prefer to extend this template, I'll be in support!

Reply:
@mariaschuld You're definitely the expert, so I'll defer to your judgment 🙂 . Which option do you think is best? I'm now thinking that your idea to make another `EntanglerLayers` template would be optimal, since it would be easier to have two separate arguments for the rotation parameters and the coupling parameters (if we are allowing for parametrized two-qubit operations).

Reply:
I'll leave it up to you. If you go with ``EntanglerLayers`` you could also make the entangler gate a custom input.

What about calling it ``CustomEntanglerLayers``? That's just because we considered removing the ``Layers`` in all templates, and it would make sense in contrast to the ``Basic`` one..

Reply:
@mariaschuld That sounds good 👍